testsys/src/run_aws_ecs.rs

```diff
@@ -209,44 +406,69 @@ impl RunAwsEcs {
+                                .clone(),
                         }
                         .into_map()
                         .context(error::ConfigMapSnafu)?,
                     ),
-                    secrets: aws_secret_map,
+                    secrets,
                     capabilities: None,
                 },
             },
             status: None,
-        };
-        let resource_client = ResourceClient::new_from_k8s_client(k8s_client.clone());
-        let test_client = TestClient::new_from_k8s_client(k8s_client);
-
-        let _ = resource_client
-            .create(ecs_resource)
-            .await
-            .context(error::ModelClientSnafu {
-                message: "Unable to create ECS cluster resource object",
-            })?;
-        println!("Created resource object '{}'", cluster_resource_name);
-
-        let _ = resource_client
-            .create(ec2_resource)
-            .await
-            .context(error::ModelClientSnafu {
-                message: "Unable to create EC2 instances resource object",
-            })?;
-        println!("Created resource object '{}'", ec2_resource_name);
-
-        let _ = test_client
-            .create(test)
-            .await
-            .context(error::ModelClientSnafu {
-                message: "Unable to create test object",
-            })?;
-        println!("Created test object '{}'", self.name);
+        })
+    }
 
-        Ok(())
+    #[allow(clippy::too_many_arguments)]
+    fn migration_test(
+        &self,
+        name: &str,
+        version: &str,
+        tuf_repo: Option<TufRepoConfig>,
+        secrets: Option<BTreeMap<String, SecretName>>,
+        ec2_resource_name: &str,
+        cluster_resource_name: &str,
+        depends_on: Option<Vec<String>>,
+        migration_agent_image: &str,
+        migration_agent_pull_secret: &Option<String>,
+    ) -> Result<Test> {
+        let mut migration_config = MigrationConfig {
+            aws_region: format!("${{{}.region}}", cluster_resource_name),
+            instance_ids: Default::default(),
+            migrate_to_version: version.to_string(),
+            tuf_repo,
+        }
+        .into_map()
+        .context(error::ConfigMapSnafu)?;
+        migration_config.insert(
+            "instanceIds".to_string(),
+            Value::String(format!("${{{}.ids}}", ec2_resource_name)),
+        );
+        Ok(Test {
+            metadata: ObjectMeta {
+                name: Some(name.to_string()),
+                namespace: Some(NAMESPACE.into()),
+                ..Default::default()
+            },
+            spec: TestSpec {
+                resources: vec![
+                    ec2_resource_name.to_owned(),
+                    cluster_resource_name.to_owned(),
+                ],
+                depends_on,
+                agent: Agent {
+                    name: "ecs-test-agent".to_string(),
+                    image: migration_agent_image.to_string(),
```

Comment:
Is the agent name accurate? Would this be clearer?
```suggestion
                    name: "ecs-migration-test-agent".to_string(),
                    image: migration_agent_image.to_string(),
```

Reply:
The name here should basically match the name of the crate that produced the image. I commented on this early. The `name` field of the `Agent` model is poorly conceived. I thought of it as a name for the program, e.g.
- ecs-test-agent
- sonobuoy-test-agent
- migration-test-agent
- eks-resource-agent

I don't think we are using it for anything at all and we should probably remove it from the model since it's a source of confusion that adds no value.

Reply:
Created #306